cylc/flow/scheduler_cli.py

```diff
@@ -523,6 +523,7 @@ def _version_check(
                     process=str.lower,
                 )
             # we are in non-interactive mode, abort abort abort
+            print('Use "--upgrade" to upgrade the workflow.', file=sys.stderr)
```

Comment:
There is a bit of a mismatch printing this to stderr when just above
```
This workflow was previously run with <yellow>{last_run_version}</yellow>.
This version of Cylc is <green>{__version__}</green>.
```
was printed to stdout

Reply:
@oliver-sanders  - Any reason not to print the earlier msg to stderr?

Reply:
Unfortunately the GUI and TUI are only regurgitating stderr. We should find a way for them to include the useful info about which versions are flying about IMHO

Reply:
Fair point, but a widespread issue not related to this PR.

I don't think stdout/err are a great interface for Tui/GUI, but short of a Python API (that allows Cylc version changing) I can't think of a better option that isn't tons of work.

In this case, the version compat issue *should* also be handled better in at the UI end as we intend to implement upgrade logic there.

Reply:
Moved to #5554